client/my-sites/plugins/controller-logged-out.js

```diff
@@ -36,11 +41,43 @@ const prefetchPopularPlugins = ( queryClient, options ) =>
 const prefetchFeaturedPlugins = ( queryClient ) =>
 	prefetchPluginsData( queryClient, getWPCOMFeaturedPluginsQueryParams() );
 
-const prefetchProductList = ( store ) => {
-	const productsList = getProductsList( store.getState() );
-	if ( Object.values( productsList ).length === 0 ) {
-		return requestProductsList( { type: 'all' } )( store.dispatch );
+const prefetchProductList = ( queryClient, store ) => {
+	const type = 'all';
+
+	return queryClient
+		.fetchQuery( [ 'products-list', type ], () => wpcom.req.get( '/products', { type } ) )
+		.then( ( productsList ) => {
+			return store.dispatch( receiveProductsList( productsList, type ) );
+		} );
+};
+
+const prefetchTimebox = ( prefetchPromises, context, timeout ) => {
+	const racingPromises = [ Promise.all( prefetchPromises ) ];
+	const isBot = context.res?.req?.useragent?.isBot;
+
+	if ( ! timeout ) {
+		timeout = isBot ? PREFETCH_TIMEOUT_BOTS : PREFETCH_TIMEOUT;
 	}
+
+	if ( config.isEnabled( 'ssr/prefetch-timebox' ) ) {
+		const timeboxPromise = new Promise( ( _, reject ) =>
+			setTimeout( reject, timeout, new Error( PREFETCH_TIMEOUT_ERROR ) )
```

Comment:
nit: we could remove `timeout` from the function signature and just move the `isBot ? ...` logic inline, since `timeout` is never passed to the function

Reply:
That's a fair point. It's probably not worth optimizing for something we don't need yet.
If we'll need different timeouts for other prefetch batches, we might as well add the timeout parameter then.